### ISSUE TITLE
SmartCollection.UpdateProductOrderAsync incomplete implementation

### ISSUE THREAD

**Issue by foxandrewj** [nozzlegear/ShopifySharp#245]
I went to use the UpdateProductOrderAsync and found that it was throwing an error, and the Unit Test was incomplete,
I found the following issues:
- The API call does not return the SmartCollection, just an empty response (like a delete)
- An additional parameter "sortOrder" is also available.  And an important one since the order of a SmartCollection cannot be applied unless the sortOrder = "manual".

I have implemented the changes and added a Unit Test and will try to submit a Pull Request soon for your review.

**Comment by nozzlegear** [comment]
Huh that's weird, I usually don't merge a PR without having full unit tests. Must have missed it!




--


Josh Harms







On Fri, May 11, 2018 at 2:35 PM -0500, "Andrew Fox" <notifications@github.com> wrote:












I went to use the UpdateProductOrderAsync and found that it was throwing an error, and the Unit Test was incomplete,

I found the following issues:

The API call does not return the SmartCollection, just an empty response (like a delete)
An additional parameter "sortOrder" is also available.  And an important one since the order of a SmartCollection cannot be applied unless the sortOrder = "manual".



I have implemented the changes and added a Unit Test and will try to submit a Pull Request soon for your review.
